### PR TITLE
Fix header menu width alignment and padding

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
- Adjusted padding styles to match page layout
- Fix focus styles on ProductNavigationLink, it got unwanted focus styles when clicking the item

### Before
https://github.com/user-attachments/assets/ca4c7641-142b-4276-a4fd-dc46c98aa09a

### After
https://github.com/user-attachments/assets/59aced56-e057-466b-8498-5e9a3636b3fb

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
* **#4505** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4505?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/HeaderNew/Header.css.ts [hunk 1]
```diff
@@ -87,7 +86,7 @@ export const menuWrapper = style({
   justifyContent: 'space-between',
   alignItems: 'center',
   flex: 1,
-  gap: tokens.space.xs,
+  paddingInline: tokens.space.md,
```
Comment (gustaveen) [inline apps/store/src/components/HeaderNew/Header.css.ts:89]: Padding instead of gap o make sure focus styles aren't clipped

Comment (alebedev) [reply apps/store/src/components/HeaderNew/Header.css.ts:89]: Also, larger clickable area is always good

**Review by alebedev — APPROVED** [review]
Added via Giphy

**Pull request merged**